source/agora/test/ValidatorCleanRestart.d

```diff
@@ -0,0 +1,149 @@
+/*******************************************************************************
+
+    Contains tests for general situations where validators participate in
+    nominating a block and reaching consensus.
+
+    Copyright:
+        Copyright (c) 2020 BOS Platform Foundation Korea
+        All rights reserved.
+
+    License:
+        MIT License. See LICENSE for details.
+
+*******************************************************************************/
+
+module agora.test.ValidatorCleanRestart;
+
+version (unittest):
+
+import agora.api.FullNode;
+import agora.common.crypto.Key;
+import agora.consensus.data.Enrollment;
+import agora.consensus.data.Transaction;
+import agora.test.Base;
+
+/// Situation: One set of Validators(set A) which enrolled in the Genesis
+///     block are all expired at the latest block height. And the other
+///     set of Validators(set B) are already enrolled and all validators.
+///     But all the set B has been shutdown and lost their data, which
+///     means that they have to have a catch-up process. After the catup-up
+///     processes of the set B have finished, a new block is being nominated,
+///     and a consensus process for the new block is being made.
+/// Expectation: The new block is approved and inserted into the ledger.
+unittest
+{
+    const validator_count = 4;
+    const outsider_count = 4;
```

Comment:
The reason we use named constant is to make things clearer. I'm not quite sure why they are used here though ?